### ISSUE TITLE
unicorn/explicit-length-check: false positive on string properties named 'size'

### ISSUE THREAD

**Issue by privatenumber** [privatenumber/lintroll#89]
## Problem

The `unicorn/explicit-length-check` rule incorrectly flags string properties that happen to be named `size`.

**Code:**
```ts
const size = props.size || 'medium'
```

**Error:**
> Use `.size > 0` when checking size is not zero

**Context:** `props.size` is a string property (e.g., `'small' | 'medium' | 'large'`), not a collection's `.size` property. The rule assumes any property named `size` is a numeric length check.

## Upstream

This is a known issue: [sindresorhus/eslint-plugin-unicorn#1266](https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1266)

## Suggested Fix

Options:
1. Disable the rule entirely in lintroll
2. Wait for upstream fix that adds type-awareness
3. Configure rule options if any exist to reduce false positives

Given lintroll's opinionated nature, disabling might be the pragmatic choice until upstream improves type inference.
